lib/puppet/type/cisco_ospf_area_vlink.rb

```diff
@@ -0,0 +1,248 @@
+      dead_interval                      => 500,
+      hello_interval                     => 2000,
+      message_digest_algorithm_type      => 'md5',
+      message_digest_encryption_type     => cisco_type_7,
+      message_digest_key_id              => 123,
+      message_digest_password            => '12345678901234567890',
+      retransmit_interval                => 777,
+      transmit_delay                     => 333,
+    }
+  "
+
+  ensurable
+
+  ###################
+  # Resource Naming #
+  ###################
+
+  # Parse out the title to fill in the attributes in these
+  # patterns. These attributes can be overwritten later.
+  def self.title_patterns
+    identity = ->(x) { x }
+    patterns = []
+
+    # Below pattern matches both parts of the full composite name.
+    patterns << [
+      /^(\S+) (\S+) (\S+) (\S+)$/,
+      [
+        [:ospf, identity],
+        [:vrf, identity],
+        [:area, identity],
+        [:vlink, identity],
+      ],
+    ]
+    patterns
+  end
+
+  # Overwrites name method. Original method simply returns self[:name],
+  # which is no longer valid or complete.
+  # Would not have failed, but just return nothing useful.
+  def name
+    "#{self[:ospf]} #{self[:vrf]} #{self[:area]} #{self[:vlink]}"
+  end
+
+  newparam(:name) do
+    desc 'Name of cisco_ospf_area_vlink, not used, but needed for puppet'
+  end
+
+  newparam(:vlink, namevar: true) do
+    desc 'Name of the virtual_link instance. Valid values are string.'
+  end # param vlink
+
+  newparam(:area, namevar: true) do
+    desc 'Name of the resource instance. Valid values are string.'
+    munge do |value|
+      value = IPAddr.new(value.to_i, Socket::AF_INET) unless
+        value[/\./]
+      value
+    end
+  end # param area
+
+  newparam(:vrf, namevar: true) do
+    desc "Name of the vrf instance. Valid values are string. The
+          name 'default' is a valid VRF."
+  end # param vrf
+
+  newparam(:ospf, namevar: true) do
+    desc 'Name of the ospf instance. Valid values are string.'
+  end # param ospf
+
+  ##############
+  # Attributes #
+  ##############
+
+  newproperty(:auth_key_chain) do
+    desc "Authentication password key chain name. Valid
+          values are string, keyword 'default'. "
+
+    munge { |value| value == 'default' ? :default : value }
+  end # property auth_key_chain
+
+  newproperty(:authentication) do
+    desc 'Enable authentication for the virtual_link.'
+
+    newvalues(:cleartext, :md5, :null, :default)
+  end # property authentication
+
+  newproperty(:authentication_key_encryption_type) do
+    desc "Specifies the scheme used for encrypting
+          authentication key password. Valid values are
+          'cleartext', '3des' or 'cisco_type_7' encryption,
+          and 'default', which defaults to 'cleartext'."
+
+    newvalues(:cleartext,
+              :'3des',
+              :cisco_type_7,
+              :default)
+
+    munge do |value|
+      value = :cleartext if value.to_sym == :default
```

Comment:
This seems odd. Why is it necessary to convert default? Why not just do:  ` munge(&:to_sym)`  ?